### PR TITLE
haskellPackages.stack: enableSharedExecutables = false;

### PR DESCRIPTION
**Pull request by anderspapitto**
The `stack` binary has a large number of dependencies, and therefore
loads a large number of shared objects on startup (I count 20,000 `open`
syscalls with `strace` when running `stack --version`).

Statically linking haskell dependencies brings the time taken for `stack
--version` from 1.3 seconds to 0.006 seconds on my machine. I think it's
worth it, since the binary is frequently invoked for short operations
during a dev workflow.

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -157140,6 +157140,7 @@ self: {
          QuickCheck resourcet retry smallcheck store template-haskell
          temporary text th-reify-many transformers vector
        ];
+       enableSharedExecutables = false;
```
Comment (anderspapitto) [inline pkgs/development/haskell-modules/hackage-packages.nix]: this is the only actual change, the rest is just automatic trailing whitespace removal (feel free to disregard it of course)

**Comment by anderspapitto** [comment]
note: upstream ships `stack` as a statically linked binary. https://docs.haskellstack.org/en/stable/ChangeLog/

**Comment by anderspapitto** [comment]
@peti what do you think? (the reviewer-adding bot seems to have not kicked in here)

**Comment by anderspapitto** [comment]
note: code diff with whitespace changes ignored: https://github.com/NixOS/nixpkgs/pull/20133/files?w=1

**Comment by peti** [comment]
We've been building a lean version of `stack` for quite some time: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L4899-L4904.

**Comment by themoritz** [comment]
When I install `haskellPackages.stack`, I get a dynamically linked executable and with it the slow startup times.

@anderspapitto Do you know what I need to do to get a statically linked one via nix? (ATM I'm manually downloading the statically linked executable from the stack github release page.)

**Comment by puffnfresh** [comment]
@themoritz you want `stack` not `haskellPackages.stack`.

**Comment by themoritz** [comment]
Awesome, thanks!

**Pull request closed without merging**